extensions/ql-vscode/src/data-extensions-editor/data-extensions-editor-view.ts

```diff
@@ -380,13 +385,54 @@ export class DataExtensionsEditorView extends AbstractWebview<
       let predictedModeledMethods: Record<string, ModeledMethod>;
 
       if (useLlmGenerationV2()) {
+        // Generate a qlpack with a filter that only includes the usages we want to model.
+        const packDir = (await dir({ unsafeCleanup: true })).path;
+
+        const syntheticConfigPack = {
+          name: "codeql/automodel-filter",
+          version: "0.0.0",
+          library: true,
+          extensionTargets: {
+            [`codeql/${this.databaseItem.language}-all`]: "*",
+          },
+          dataExtensions: ["filter.yml"],
+        };
+
+        const qlpackFile = join(packDir, "codeql-pack.yml");
+        await outputFile(qlpackFile, dumpYaml(syntheticConfigPack), "utf8");
+
+        // TODO: this filter is just static as an example. We want to generate this from the usages.
+        const filter = {
+          extensions: [
+            {
+              addsTo: {
+                pack: `codeql/${this.databaseItem.language}-queries`,
+                extensible: "automodelCandidateFilter",
+              },
+              data: [
+                ["org.sql2o", "Sql2o", "open", "()"],
+                [
+                  "org.springframework.boot",
+                  "SpringApplication",
+                  "run",
+                  "(Class,String[])",
+                ],
+              ],
+            },
+          ],
+        };
+
+        const filterFile = join(packDir, "filter.yml");
+        await writeFile(filterFile, dumpYaml(filter), "utf8");
+
         const usages = await runAutoModelQueries({
           mode: this.mode,
           cliServer: this.cliServer,
           queryRunner: this.queryRunner,
           queryStorageDir: this.queryStorageDir,
           databaseItem: this.databaseItem,
           progress: (update) => progress({ ...update, maxStep }),
+          extraExtensionPacks: [packDir],
```

Comment:
I was expecting to see the name of the extension pack here. I haven't looked at the entire implementation, so you may already be doing this properly. The `--extension-packs` argument must be the name (and optional version) of the extension/model pack you want to use. And this pack must be available either in the package cache or from addtional-packs or from the workspace.